### PR TITLE
Improve unit tests

### PR DESCRIPTION
**Pull request by lmbarros**
The new tests make use of some deltas and signatures generated by the
rdiff from the original librsync (the one written in C). The idea of
these tests is to make sure we are generating results that match those
of the original library.

Some of the test cases are disabled because they currently fail. They will be enabled in upcoming PR that fix these issues.

### CODE REVIEW

File: testdata/README.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# librsync-go Test Data
+
+Reference files (`*.signature`, `*.delta`) were created using the original (C
```
Comment (robertgzr) [inline testdata/README.md:3]: Is it worth it to have a script/function that could be used to generate the test data with `rdiff`? in the spirit of the stdlib's golden files https://github.com/golang/go/blob/master/src/cmd/internal/test2json/test2json_test.go#L21

I guess it's not necessary to be done again right?

Comment (lmbarros) [reply testdata/README.md]: Hum, good question. We should not need to generate _these_ files again but someone probably will  want to add more test files in the future. I added one more commit explaining how to I ran `rdiff` to generate them.

**Comment by robertgzr** [comment]
@balena-ci rebase

File: testdata/README.md (hunk 1 above)
Comment (robertgzr) [reply testdata/README.md]: nice o/

**Comment by robertgzr** [comment]
@balena-ci rebase

File: signature.go [hunk 2]
```diff
@@ -96,3 +97,62 @@ func Signature(input io.Reader, output io.Writer, blockLen, strongLen uint32, si
 
 	return &ret, nil
 }
+
+// ReadSignatureFile reads a signature from the file at path.
+func ReadSignatureFile(path string) (*SignatureType, error) {
```
Comment (robertgzr) [inline signature.go:102]: How about exposing two functions? one that takes a `io.Reader` and another one that wraps it and does the `os.Open` call on a path. That gives full flexibility to any library user. Like if we decide to cache signatures at some point :)

Comment (lmbarros) [reply signature.go]: Yep, good suggestion, thanks! Looks more like proper Go :-)

Just pushed an updated version!

**Review by robertgzr — APPROVED** [review]
Beautiful :)

```
#13 4.561 === RUN   TestDeltaSmokeTest
#13 4.561 --- PASS: TestDeltaSmokeTest (0.03s)
#13 4.561 === RUN   TestPatch
#13 4.561 === RUN   TestPatch/000-blake2-11-23
#13 4.561 === RUN   TestPatch/000-blake2-512-32
#13 4.561 === RUN   TestPatch/000-md4-256-7
#13 4.561 === RUN   TestPatch/001-blake2-512-32
#13 4.561 === RUN   TestPatch/001-blake2-776-31
#13 4.561 === RUN   TestPatch/001-md4-777-15
#13 4.561 === RUN   TestPatch/002-blake2-512-32
#13 4.561 === RUN   TestPatch/002-blake2-431-19
#13 4.561 === RUN   TestPatch/002-md4-128-16
#13 4.561 === RUN   TestPatch/003-blake2-512-32
#13 4.561 === RUN   TestPatch/003-blake2-1024-13
#13 4.561 === RUN   TestPatch/003-md4-1024-13
#13 4.561 === RUN   TestPatch/004-blake2-1024-28
#13 4.561 === RUN   TestPatch/004-blake2-2222-31
#13 4.561 === RUN   TestPatch/004-blake2-512-32
#13 4.561 === RUN   TestPatch/005-blake2-512-32
#13 4.561 === RUN   TestPatch/006-blake2-2-32
#13 4.561 === RUN   TestPatch/007-blake2-4-32
#13 4.561 === RUN   TestPatch/007-blake2-3-32
#13 4.561 === RUN   TestPatch/009-blake2-2048-26
#13 4.561 === RUN   TestPatch/009-blake2-512-32
#13 4.561 === RUN   TestPatch/009-md4-2033-15
#13 4.561 --- PASS: TestPatch (0.00s)
#13 4.561     --- PASS: TestPatch/000-blake2-11-23 (0.00s)
#13 4.561     --- PASS: TestPatch/000-blake2-512-32 (0.00s)
#13 4.561     --- PASS: TestPatch/000-md4-256-7 (0.00s)
#13 4.561     --- PASS: TestPatch/001-blake2-512-32 (0.00s)
#13 4.561     --- PASS: TestPatch/001-blake2-776-31 (0.00s)
#13 4.561     --- PASS: TestPatch/001-md4-777-15 (0.00s)
#13 4.561     --- PASS: TestPatch/002-blake2-512-32 (0.00s)
#13 4.561     --- PASS: TestPatch/002-blake2-431-19 (0.00s)
#13 4.561     --- PASS: TestPatch/002-md4-128-16 (0.00s)
#13 4.561     --- PASS: TestPatch/003-blake2-512-32 (0.00s)
#13 4.561     --- PASS: TestPatch/003-blake2-1024-13 (0.00s)
#13 4.561     --- PASS: TestPatch/003-md4-1024-13 (0.00s)
#13 4.561     --- PASS: TestPatch/004-blake2-1024-28 (0.00s)
#13 4.561     --- PASS: TestPatch/004-blake2-2222-31 (0.00s)
#13 4.561     --- PASS: TestPatch/004-blake2-512-32 (0.00s)
#13 4.561     --- PASS: TestPatch/005-blake2-512-32 (0.00s)
#13 4.561     --- PASS: TestPatch/006-blake2-2-32 (0.00s)
#13 4.561     --- PASS: TestPatch/007-blake2-4-32 (0.00s)
#13 4.561     --- PASS: TestPatch/007-blake2-3-32 (0.00s)
#13 4.561     --- PASS: TestPatch/009-blake2-2048-26 (0.00s)
#13 4.561     --- PASS: TestPatch/009-blake2-512-32 (0.00s)
#13 4.561     --- PASS: TestPatch/009-md4-2033-15 (0.00s)
#13 4.561 === RUN   TestDeltaAndPatch
#13 4.561 === RUN   TestDeltaAndPatch/000-blake2-11-23
#13 4.561 === RUN   TestDeltaAndPatch/000-blake2-512-32
#13 4.561 === RUN   TestDeltaAndPatch/000-md4-256-7
#13 4.561 === RUN   TestDeltaAndPatch/001-blake2-512-32
#13 4.561 === RUN   TestDeltaAndPatch/001-blake2-776-31
#13 4.561 === RUN   TestDeltaAndPatch/001-md4-777-15
#13 4.561 === RUN   TestDeltaAndPatch/002-blake2-512-32
#13 4.561 === RUN   TestDeltaAndPatch/002-blake2-431-19
#13 4.561 === RUN   TestDeltaAndPatch/002-md4-128-16
#13 4.561 === RUN   TestDeltaAndPatch/003-blake2-512-32
#13 4.561 === RUN   TestDeltaAndPatch/003-blake2-1024-13
#13 4.561 === RUN   TestDeltaAndPatch/003-md4-1024-13
#13 4.561 === RUN   TestDeltaAndPatch/004-blake2-1024-28
#13 4.561 === RUN   TestDeltaAndPatch/004-blake2-2222-31
#13 4.561 === RUN   TestDeltaAndPatch/004-blake2-512-32
#13 4.561 === RUN   TestDeltaAndPatch/005-blake2-512-32
#13 4.561 === RUN   TestDeltaAndPatch/006-blake2-2-32
#13 4.561 === RUN   TestDeltaAndPatch/007-blake2-4-32
#13 4.561 === RUN   TestDeltaAndPatch/007-blake2-3-32
#13 4.561 === RUN   TestDeltaAndPatch/009-blake2-2048-26
#13 4.561 === RUN   TestDeltaAndPatch/009-blake2-512-32
#13 4.561 === RUN   TestDeltaAndPatch/009-md4-2033-15
#13 4.561 --- PASS: TestDeltaAndPatch (0.04s)
#13 4.561     --- PASS: TestDeltaAndPatch/000-blake2-11-23 (0.00s)
#13 4.561     --- PASS: TestDeltaAndPatch/000-blake2-512-32 (0.00s)
#13 4.561     --- PASS: TestDeltaAndPatch/000-md4-256-7 (0.00s)
#13 4.561     --- PASS: TestDeltaAndPatch/001-blake2-512-32 (0.00s)
#13 4.561     --- PASS: TestDeltaAndPatch/001-blake2-776-31 (0.03s)
#13 4.561     --- PASS: TestDeltaAndPatch/001-md4-777-15 (0.00s)
#13 4.561     --- PASS: TestDeltaAndPatch/002-blake2-512-32 (0.00s)
#13 4.561     --- PASS: TestDeltaAndPatch/002-blake2-431-19 (0.00s)
#13 4.561     --- PASS: TestDeltaAndPatch/002-md4-128-16 (0.00s)
#13 4.561     --- PASS: TestDeltaAndPatch/003-blake2-512-32 (0.00s)
#13 4.561     --- PASS: TestDeltaAndPatch/003-blake2-1024-13 (0.00s)
#13 4.561     --- PASS: TestDeltaAndPatch/003-md4-1024-13 (0.00s)
#13 4.561     --- PASS: TestDeltaAndPatch/004-blake2-1024-28 (0.00s)
#13 4.561     --- PASS: TestDeltaAndPatch/004-blake2-2222-31 (0.00s)
#13 4.561     --- PASS: TestDeltaAndPatch/004-blake2-512-32 (0.00s)
#13 4.561     --- PASS: TestDeltaAndPatch/005-blake2-512-32 (0.00s)
#13 4.561     --- PASS: TestDeltaAndPatch/006-blake2-2-32 (0.00s)
#13 4.561     --- PASS: TestDeltaAndPatch/007-blake2-4-32 (0.00s)
#13 4.561     --- PASS: TestDeltaAndPatch/007-blake2-3-32 (0.00s)
#13 4.561     --- PASS: TestDeltaAndPatch/009-blake2-2048-26 (0.00s)
#13 4.561     --- PASS: TestDeltaAndPatch/009-blake2-512-32 (0.00s)
#13 4.561     --- PASS: TestDeltaAndPatch/009-md4-2033-15 (0.00s)
#13 4.561 === RUN   TestRollsumBrandNew
#13 4.561 --- PASS: TestRollsumBrandNew (0.00s)
#13 4.561 === RUN   TestRollsumRollinRollout
#13 4.561 --- PASS: TestRollsumRollinRollout (0.00s)
#13 4.561 === RUN   TestRollsumUpdate
#13 4.561 --- PASS: TestRollsumUpdate (0.00s)
#13 4.561 === RUN   TestRollsumRotate
#13 4.561 --- PASS: TestRollsumRotate (0.00s)
#13 4.561 === RUN   TestRollsumConsistency
#13 4.561 --- PASS: TestRollsumConsistency (0.00s)
#13 4.561 === RUN   TestRollsumRotateByteSubtractionBug
#13 4.561 --- PASS: TestRollsumRotateByteSubtractionBug (0.00s)
#13 4.561 === RUN   TestSignature
#13 4.561 === RUN   TestSignature/003-blake2-512-32
#13 4.561 === RUN   TestSignature/003-blake2-1024-13
#13 4.561 === RUN   TestSignature/003-md4-1024-13
#13 4.561 === RUN   TestSignature/007-blake2-5-32
#13 4.561 === RUN   TestSignature/009-blake2-2048-26
#13 4.561 === RUN   TestSignature/009-blake2-512-32
#13 4.561 === RUN   TestSignature/009-md4-2033-15
#13 4.561 === RUN   TestSignature/010-blake2-512-32
#13 4.561 === RUN   TestSignature/010-blake2-7-6
#13 4.561 === RUN   TestSignature/010-md4-4096-8
#13 4.561 --- PASS: TestSignature (0.00s)
#13 4.561     --- PASS: TestSignature/003-blake2-512-32 (0.00s)
#13 4.561     --- PASS: TestSignature/003-blake2-1024-13 (0.00s)
#13 4.561     --- PASS: TestSignature/003-md4-1024-13 (0.00s)
#13 4.561     --- PASS: TestSignature/007-blake2-5-32 (0.00s)
#13 4.561     --- PASS: TestSignature/009-blake2-2048-26 (0.00s)
#13 4.561     --- PASS: TestSignature/009-blake2-512-32 (0.00s)
#13 4.561     --- PASS: TestSignature/009-md4-2033-15 (0.00s)
#13 4.561     --- PASS: TestSignature/010-blake2-512-32 (0.00s)
#13 4.561     --- PASS: TestSignature/010-blake2-7-6 (0.00s)
#13 4.561     --- PASS: TestSignature/010-md4-4096-8 (0.00s)
#13 4.561 PASS
#13 4.561 coverage: 82.2% of statements
#13 4.561 ok  	github.com/balena-os/librsync-go	0.073s	coverage: 82.2% of statements
#13 4.561 ?   	github.com/balena-os/librsync-go/cmd/rdiff	[no test files]
#13 DONE 4.8s
```

**Pull request merged**